packages/matchbox/src/components/TextField/TextField.js

```diff
@@ -1,179 +1,207 @@
+    height: multiline ? 'auto' : '2.5rem',
+    name,
+    id,
+    type,
+    disabled,
+    readOnly,
+    autoFocus,
+    placeholder,
+    onFocus,
+    onBlur,
+    onChange,
+    pl: prefixPadding,
+    pr: suffixPadding,
+    textAlign: align,
+    ...describedBy,
+    style: {
+      ...(multiline ? { resize } : {}),
+      ...style,
+    },
+  };
 
-    const labelMarkup = (
-      <Label id={id} label={`${label}${requiredIndicator}`}>
-        {error && errorInLabel && <Error className={styles.InlineError} wrapper='span' error={error} />}
+  return (
+    <StyledWrapper {...systemProps}>
+      <Label id={id} label={label} labelHidden={labelHidden}>
+        {required && (
+          <Box as="span" pr="200" aria-hidden="true">
+            *
+          </Box>
+        )}
+        {error && errorInLabel && (
+          <Box as={Error} id={errorId} wrapper="span" error={error} fontWeight="400" />
+        )}
       </Label>
-    );
-
-    const helpMarkup = helpText
-      ? <div className={styles.HelpText}>{helpText}</div>
-      : null;
-
-    const prefixMarkup = prefix
-      ? <span className={prefixClasses} ref={(node) => this.prefixNode = node}>{prefix}</span>
-      : null;
-
-    const suffixMarkup = suffix
-      ? <span className={suffixClasses} ref={(node) => this.suffixNode = node}>{suffix}</span>
-      : null;
-
-    const input = React.createElement(multiline ? 'textarea' : 'input', {
-      ...rest,
-      name,
-      id,
-      type,
-      disabled,
-      readOnly,
-      autoFocus,
-      placeholder,
-      onFocus,
-      onBlur,
-      onChange,
-      className: setInputClasses,
-      style: {
-        textAlign: align,
-        paddingLeft,
-        paddingRight,
-        resize,
-        ...style
-      }
-    });
-
-    const inputMarkup = prefix || suffix
-      ? <div className={styles.InputWrapper}>
-        {prefixMarkup}
-        {input}
-        {suffixMarkup}
-      </div>
-      : input;
-
-    return (
-      <fieldset className={styles.TextField}>
-        {label && !labelHidden && labelMarkup}
-        <Connect left={connectLeft} right={connectRight}>
-          {inputMarkup}
-        </Connect>
-        {error && !errorInLabel && <Error error={error} />}
-        {helpMarkup}
-      </fieldset>
-    );
-  }
+      <Connect left={connectLeft} right={connectRight}>
+        <Box position="relative">
+          <PrefixOrSuffix
+            content={prefix}
+            className={prefixClassname}
+            forwardedRef={prefixRef}
+            left="300"
+          />
+          <FieldBox {...inputProps} />
+          <PrefixOrSuffix
+            content={suffix}
+            className={suffixClassname}
+            forwardedRef={suffixRef}
+            right="300"
+          />
+        </Box>
+      </Connect>
+      {error && !errorInLabel && <Error id={errorId} error={error} />}
+      {helpText ? <HelpText id={helpTextId}>{helpText}</HelpText> : null}
```

Comment:
I think you could just do `{helpText && <HelpText/>}` rather than the ternary here